### DISCUSSION TITLE
Auto Playlist recently added track number problem

### DISCUSSION THREAD

**Discussion by giorgiobrit** [TT-ReBORN/Georgia-ReBORN#121] [Q&A] [answered]
I created an auto playlist of new songs added to the library using this syntax %ADDED% DURING LAST 6 WEEKS SORT DESCENDING BY %ADDED%. The problem is that the tracks from albums are not being ordered correctly. Some are in order, but most of them are not. I don't understand why! I have tried many experiments, but nothing works. I need your valuable help. Also, another question, is it possible to use a parameter instead of %added% that searches for tracks that are truly new based on the date? Otherwise, every time I change the library location, it adds the entire library as new songs. Thanks in advance."
![screenshot 1](https://github.com/TT-ReBORN/Georgia-ReBORN/assets/57190501/06f4ffc6-ca8d-4620-8375-51bf3b999edd)

**Comment by regorxxx** [comment, marked as answer]
Query + Sorting:
`(%added% DURING LAST 6 WEEKS AND %path% HAS d:\music) SORT DESCENDING BY "$date(%added%)" - %album% - %discnumber% - %tracknumber% - %title%`

Or

Query: `%added% DURING LAST 6 WEEKS AND %path% HAS d:\music`
Sort: `$sub(9999999,$year(%added%))|$sub(9999999,$month(%added%))|$sub(9999999,$day_of_month(%added%))`

If you want hour precision, continue the pattern. If you only neeed per year, or month, remove some part.

If you want album grouping in top of that:
Query: `...`
Sort: `$sub(9999999,$year(%added%))|$sub(9999999,$month(%added%))|$sub(9999999,$day_of_month(%added%))|%album% - %discnumber% - %tracknumber% - %title%`

If you want natural sorting:
Query: `...`
Sort: `$year(%added%)|$month(%added%)|$day_of_month(%added%)|%album% - %discnumber% - %tracknumber% - %title%`

Or 

Query + Sorting:
`(%added% DURING LAST 6 WEEKS AND %path% HAS d:\music) SORT BY "$date(%added%)" - %album% - %discnumber% - %tracknumber% - %title%`

**Reply by TT-ReBORN** [reply]
Thanks @regorxxx =)
